src/locales/scripts/read-i18n.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * In `en.json` we use a limited version of JSON5. It is valid JSON5 but only
+ * uses a limited set of features and syntax. This package provides functions
+ * to read this format and convert it into regular JSON.
+ *
+ * - All values must be strings. This is okay because we're only dealing with
+ *   i18n translation files.
+ * - Only single line `// ...` comments are allowed. These comments describe the
+ *   content that follows it. Inline comments are not allowed.
+ *
+ * ```json5
+ * {
+ *   // documentation about `key-a`
+ *   key: 'value',
+ *   'key-b': {
+ *     'key-c': 'value-c',
+ *   },
+ * }
+ * ```
+ *
+ * @typedef {{ [key: string]: string | SimJson }} SimJson
+ */
```

Comment:
This module is clever. Reading it, however, it struck me that you've essentially re-written a very basic version of a JavaScript AST parser. We could use instead use an existing AST parser and treat the `json5` file as regular JavaScript and get the following benefits:

1. No need to maintain a file parser
2. Typed AST
3. Support for multi-line comments

`@babel/parser` does a great job at all of these and is thoroughly documented. Check out this AST Explorer example: https://astexplorer.net/#/gist/fd2baa6817ccaa87c23704ead7d3e7f8/fb97d60a32d361ab1da42d9a19c88a2b22762733

You'll notice the object had to be wrapped in parens. This could be done in code by opening the file and adding an open and closing paren to the object. Alternatively, we could write the file as plain JavaScript and have it like so:

```js
module.exports = {
  // ... the translation object
```

Instead of using the JSON5 extension but only supporting a limited subset of it.

If you click on the object properties in the AST explorer link, you'll see a `leadingComments` array that includes all the leading comments. Navigating through nested blocks is also easy, thanks to the uniform AST. We could transform this into the `Entry` AST for simplicity’s sake and to pare down the unnecessary parts of the babel AST: the goal of using babel's parser is mostly to avoid having to maintain our own AST parser :slightly_smiling_face: 

Finally, for generating the `en.json`, if we use `json5` we can use the `json5` reference implementation to get the comment-free version of the object. If we use JavaScript for the file instead, then we can just import the object and write it out as JSON without the need for an additional package.

None of these suggestions are blocking, but I am hesitant to approve without further clarification for why we should add a simplified JSON5 parser to our codebase.

Reply:
I searched everywhere for a JSON-AST package but didn't find any that supported JSON5 and had been actively maintained. The JSON5 parsers I found didn't have the option for reading to AST, only parsing it which strips all comments.

I hadn't considered the idea of treating the file as JS and using `@babel/parser` on it. That might be worth investigating for the consistency of having a standard AST, although
- the rudimentary parser is only a _tiny bit_ of a maintenance burden because it's only a bunch of `if` statements
- Babel's parser is overkill for a JSON file and then getting the information we want from the AST it outputs will also add a fair bit of code

For generating the `en.json` I didn't want to use the `json5` parser because we already have all the information we need to write the JSON in the `Entry` object so it made sense to me to just use that to output the JSON.

To be honest, I'm being biased towards the hack parser because I spent a lot of time writing and testing it but I'm open to using a full-fledged parser like Babel if it improves the quality of the PR.

Reply:
I refactored `read-i18n.js` to use the `@babel/parser` package and it while the LoC didn't change much, the code looks much cleaner. Thanks again for the suggestion, wish I had asked you for pointers sooner!